### PR TITLE
Refactor and modernise UI

### PR DESCRIPTION
**Pull request by adamgundry**
This refactors some of the internal code and makes a variety of UI tweaks:
 * Tabs now use Bootstrap (added a dependency on `popper.js`)
 * The layout has been condensed
 * A new "Meta" tab shows metadata that is not relevant for every view
 * Heap profiling tabs are combined into a single dropdown
 * Tabs appear even if the underlying data is not available, with an explanation how to get the data (some of these explanations need work)
 * Tabs can have documentation associated with them; so far this exists partially for the "Heap" tab
 * When using `-hi` profiling, the main heap profile charts show the info table name (if available) in addition to the pointer
 * Ticky data is displayed in a separate tab (if available), alongside the other tabs, rather than being a separate mode

This is joint work with @adinapoli.

Screenshot

Screenshot when no data available

### CODE REVIEW

**Comment by BinderDavid** [comment]
We are sadly still blocked on #176 :( I can see whether I find the time on Sunday to rewrite the Github actions.

**Comment by TeofilC** [comment]
Wow this looks great from the screenshots!

**Comment by BinderDavid** [comment]
Looking at the log, I think CI can maybe also be fixed by just updating the cachix action: https://github.com/cachix/cachix-action/issues/144

**Comment by adamgundry** [comment]
Thanks for the prompt action on the CI @BinderDavid! Looks like CI is now green, though I don't know if there is a way to see the generated Hakyll site that it built?

**Comment by adinapoli** [comment]
> Super minor thing, but would it be possible to remove the delay/transition when switching tabs

It should be configurable via bootstrap if memory serves me well, but I will double check later this week.

File: src/Eventlog/AssetVersions.hs [hunk 1]
```diff
@@ -39,3 +39,6 @@ datatablesButtonsCSSURL = "https://cdn.datatables.net/buttons/2.0.1/css/buttons.
 datatablesButtonsHTML5Version,datatablesButtonsHTML5URL :: String
 datatablesButtonsHTML5Version = "2.0.1"
 datatablesButtonsHTML5URL = "https://cdn.datatables.net/buttons/2.0.1/js/buttons.html5.min.js"
+popperVersion,popperURL :: String
+popperVersion = "1.14.3"
+popperURL = "https://cdn.jsdelivr.net/npm/popper.js@1.14.3/dist/umd/popper.min.js"
```
Comment (fendor) [inline src/Eventlog/AssetVersions.hs:44]: Needs to be regenerated

File: src/Eventlog/Ticky.hs [hunk 2]
```diff
@@ -268,8 +180,8 @@ closureSize fvs cl_args
 
 
 
-renderSpark :: Int -> [(Double, Word64, Word64)] -> Html
-renderSpark size vs = H.span ! A.class_ "linechart"
+_renderSpark :: Int -> [(Double, Word64, Word64)] -> Html
+_renderSpark size vs = H.span ! A.class_ "linechart"
```
Comment (fendor) [inline src/Eventlog/Ticky.hs:184]: Is this unneeded now?

**Review by fendor — APPROVED** [review]
Changes look good to me, only some nitpicking changes

File: versions.json [hunk 3]
```diff
@@ -12,6 +12,5 @@
   , { "hname" : "datatablesButtons", "version" : "2.0.1", "url" : "https://cdn.datatables.net/buttons/<>/js/dataTables.buttons.min.js" }
   , { "hname" : "datatablesButtonsCSS", "version" : "2.0.1", "url" : "https://cdn.datatables.net/buttons/<>/css/buttons.dataTables.min.css" }
   , { "hname" : "datatablesButtonsHTML5", "version" : "2.0.1", "url" : "https://cdn.datatables.net/buttons/<>/js/buttons.html5.min.js" }
-
+  , { "hname" : "popper", "version" : "1.14.3", "url": "https://cdn.jsdelivr.net/npm/popper.js@1.14.3/dist/umd/popper.min.js" }
```
[the remark this answers is not in the document]
Comment (fendor) [reply versions.json:15]: I manually verified `https://cdn.jsdelivr.net/npm/popper.js@1.14.3` resolves to the same content as `https://cdn.jsdelivr.net/npm/popper.js@1.14.3/dist/umd/popper.min.js`

**Comment by fendor** [comment]
Manually validated that the gh-pages render properly for this patch

**Comment by fendor** [comment]
> It should be configurable via bootstrap if memory serves me well, but I will double check later this week.

Fixed as well.

**Review by BinderDavid — APPROVED** [review]

**Pull request merged**